test/immediate_sync_test.cpp

```diff
@@ -88,3 +88,90 @@ TEST_F(ManagerTest, testDataChangeInFile)
 
     ctx.run();
 }
+
+TEST_F(ManagerTest, testDataChangeDisableProperty)
```

Comment:
> testDataChangeDisableProperty

I think its not clear. looks you are trying to test data change when the sync is disabled/

How about `testDataChangeWhenSyncIsDisabled`?